### ISSUE TITLE
Cupertino dialogs do not have route semantics or lables

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#21250]
Each Cupertino alert dialog widget needs to include a Semantics node with scopes/names route and a label of "ALERT" (localized).

The accessibility tree currently looks like so for a cupertino alert dialog:
```
I/flutter (10104): SemanticsNode#0
I/flutter (10104):  │ Rect.fromLTRB(0.0, 0.0, 1080.0, 1794.0)
I/flutter (10104):  │
I/flutter (10104):  └─SemanticsNode#68
I/flutter (10104):    │ Rect.fromLTRB(0.0, 0.0, 411.4, 683.4) scaled by 2.6x
I/flutter (10104):    │ textDirection: ltr
I/flutter (10104):    │
I/flutter (10104):    └─SemanticsNode#113
I/flutter (10104):      │ Rect.fromLTRB(0.0, 0.0, 411.4, 683.4)
I/flutter (10104):      │ flags: scopesRoute
I/flutter (10104):      │
I/flutter (10104):      ├─SemanticsNode#114
I/flutter (10104):      │ │ Rect.fromLTRB(70.7, 297.5, 340.7, 364.5)
I/flutter (10104):      │ │ scrollExtentMin: 0.0
I/flutter (10104):      │ │ scrollPosition: 0.0
I/flutter (10104):      │ │ scrollExtentMax: 0.0
I/flutter (10104):      │ │
I/flutter (10104):      │ └─SemanticsNode#115
I/flutter (10104):      │     Rect.fromLTRB(20.0, 23.0, 250.0, 47.0)
I/flutter (10104):      │     label: "Discard draft?"
I/flutter (10104):      │     textDirection: ltr
I/flutter (10104):      │
I/flutter (10104):      └─SemanticsNode#116
I/flutter (10104):        │ Rect.fromLTRB(70.7, 364.9, 340.7, 409.9)
I/flutter (10104):        │ scrollExtentMin: 0.0
I/flutter (10104):        │ scrollPosition: 0.0
I/flutter (10104):        │ scrollExtentMax: 0.0
I/flutter (10104):        │
I/flutter (10104):        ├─SemanticsNode#117
I/flutter (10104):        │ │ Rect.fromLTRB(0.0, 0.0, 134.8, 45.0)
I/flutter (10104):        │ │ actions: tap
I/flutter (10104):        │ │
I/flutter (10104):        │ └─SemanticsNode#118
I/flutter (10104):        │     Rect.fromLTRB(0.0, 0.0, 134.8, 45.0)
I/flutter (10104):        │     actions: tap
I/flutter (10104):        │     label: "Discard"
I/flutter (10104):        │     textDirection: ltr
I/flutter (10104):        │
I/flutter (10104):        └─SemanticsNode#119
I/flutter (10104):          │ Rect.fromLTRB(135.2, 0.0, 270.0, 45.0)
I/flutter (10104):          │ actions: tap
I/flutter (10104):          │
I/flutter (10104):          └─SemanticsNode#120
I/flutter (10104):              Rect.fromLTRB(0.0, 0.0, 134.8, 45.0)
I/flutter (10104):              actions: tap
I/flutter (10104):              label: "Cancel"
I/flutter (10104):              textDirection: ltr
```

**Comment by jonahwilliams** [comment]
Additionally, the action buttons are not labeled as buttons and seem to have multiple tap actions.


To resolve the lack of announcement, the dialog needs a scopes/names route flag and a name, something like this:

```
Semantics(
  scopesRoute: true, // indicates that this subtree represents a route
  namesRoute: true, // indicates that the label of this node contributes a route name
  label: 'Alert', // name of the route.
  child: dialog
)
```


For buttons, we'll probably have to exclude the gesture detectors from semantics and then add a special handle to trigger both of them on tap.

```
Semantics(
  button: true,
  onTap: () { ... },
)
```

**Comment by mklim** [comment]
It also looks like alert dialogs can have any combination of a title and content. VoiceOver automatically reads the title if both exist, and falls back to reading the content when there is no title.

**Comment by jonahwilliams** [comment]
You can also provide a `namesRoute` tag over an existing semantics node and reuse the label:

```
Semantics(
  namesRoute: true,
  child: dialogTitle,
)
```

See the doc comment here for how the route/name is selected: https://github.com/flutter/engine/blob/master/lib/ui/semantics.dart#L398
